### ISSUE TITLE
Source.fix() applies lint fixes unconditionally, mutating clean XMIR

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1373]
## Problem

`Source.fix()` applies every lint's fix **unconditionally**, even when the input XMIR has no defects for that lint. A fix that removes/moves nodes in expectation of a defect will mutate a perfectly clean document.

## Root cause

`src/main/java/org/eolang/lints/Source.java:89-95`:

```java
private XML fix(final XML xmir) {
    XML result = xmir;
    for (final Lint lint : Source.MONO.value()) {
        result = lint.fix().apply(result);
    }
    return result;
}
```

There is no gate on `lint.defects(result)`. For example, the `unsorted-metas` fix (an identity template that also sorts metas) will reorder metas even in a document whose metas are already sorted, and will touch documents that carry no metas at all.

## Minimal example

```java
final XML before = new XMLDocument("<object><metas><meta><head>spdx</head><tail>MIT</tail></meta></metas><o name=\"x\"/></object>");
final XML after = new Source(before).fix(); // metas re-serialized/reordered even though no defect exists
```

## Expected behavior

A lint's fix should be applied only when that lint reported at least one defect on the current document (or the unconditional behavior should be documented explicitly in the javadoc of `Lint.fix()`/`Source.fix()`).

## Related

- `src/main/java/org/eolang/lints/Fix.java` — the `Fix` contract; a fix may legitimately assume a defect was present.
- Existing fix pack: `fixes/unsorted-metas/sorts-version-and-spdx.yaml`.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+686](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
